src/server/session.ts

```diff
@@ -2198,7 +2198,22 @@ namespace ts.server {
             const scriptInfo = project.getScriptInfoForNormalizedPath(file)!;
             const { startPosition, endPosition } = this.getStartAndEndPosition(args, scriptInfo);
 
-            const codeActions = project.getLanguageService().getCodeFixesAtPosition(file, startPosition, endPosition, args.errorCodes, this.getFormatOptions(file), this.getPreferences(file));
+            let codeActions: readonly CodeFixAction[];
+            try {
+                codeActions = project.getLanguageService().getCodeFixesAtPosition(file, startPosition, endPosition, args.errorCodes, this.getFormatOptions(file), this.getPreferences(file));
+            }
+            catch(e) {
+                const ls = project.getLanguageService();
+                const existingDiagCodes =
+                    flatMap(["getSyntacticDiagnostics", "getSemanticDiagnostics", "getSuggestionDiagnostics"] as const,
+                            getter => ls[getter](file))
+                        .map(d => decodedTextSpanIntersectsWith(startPosition, endPosition - startPosition, d.start!, d.length!) && d.code); // TODO: GH#18217

```

Comment:
Maybe don’t add new `TODO`s

Reply:
It's used elsewhere in this file, a reference to a catchall issue for getting rid of the `!`s that are used for `.start` and `.length`.  I don't know under what conditions it *can* be undefined, so I thought that its best to use the same comment for whenever this is resolved...

Reply:
I believe they were auto-inserted by a script when the codebase turned on `--strictNullChecks`. I believe any time you call `get*Diagnostics` with a `file` argument, the results will always have a location. But my suggestion is that we should either be confident about the non-nullness of them, or account for their nullness. (I’m fairly sure we can just be confident about these and delete the TODO.)

Reply:
I was trying to follow the code to see if it's really always defined, but that was too much; removed the comment given that both of you are sure that it's ok...